### PR TITLE
Added schemas for Reconstruction activity and ReconstructedCell entity

### LINKED ISSUE #82: Add the reconstruction and reconstructedCell schemas to the morphology module

### PR DESCRIPTION
**Pull request by srjimenez**
- schemas and test files 

Fix #82

### CODE REVIEW

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  "@context": [
+    "{{base}}/contexts/nexus/shacl/rec-shacl-20170720/v0.1.0",
+    "{{base}}/contexts/nexus/core/standards/v0.1.0",
+    "{{base}}/contexts/bbp/core/entity/v0.1.0",
+    "{{base}}/contexts/bbp/neurosciencegraph/core/v0.1.0",
+    {
+      "bbptlotsh": "{{base}}/schemas/bbp/core/typedlabeledontologyterm/v0.1.0/shapes/",
+      "bbpentsh": "{{base}}/schemas/bbp/core/entity/v0.1.0/shapes/",
+      "bbplocsh": "{{base}}/schemas/bbp/core/brainlocation/v0.1.0/shapes/",
+      "this": "{{base}}/schemas/bbp/experiment/reconstructedcell/v0.1.0/shapes/"
```
Comment (MFSY) [inline modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: "experiment" or "morphology" domain ?

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+              "path" : "nsg:distanceToBoundary",
+              "name" : "Distance to boundary",
+              "description" : "Distance of the reconstructed cell to pia or to another brain boundary.",
+              "node" : "bbplocsh:DistanceToBoundaryShape",
+              "maxCount" : 1
+            },
+            {
+              "path" : "nsg:mType",
+              "name" : "Morphology type",
+              "description" : "The morphology type of the reconstructed cell: need the cell type for class restriction.",
```
Comment (MFSY) [inline modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: I suggest to remove ": need the cell type for class restriction" and to use editorialNote property for editorial comments.

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+              "node": "bbptlotsh:CellTypeOntologyTermShape",
+              "maxCount" : 1
+            },
+            {
+              "path" : "nsg:mSubType",
```
Comment (MFSY) [inline modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: What is the difference between mType and mSubType ? Is a mSubType a more specific mType ? 
If would suggest to use the same strategy as for brainRegion  and subRegion: one property called mType and the mType and mSubType are terms from the cell type ontology.

Comment (MFSY) [inline modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: I would suggest to merge this property with the nsg:mType one.

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+              "name" : "Morphology sub-type",
+              "description" : "The morphology sub-type of the reconstructed cell: need the cell sub-type for class restriction.",
+              "datatype": "xsd:string",
+              "maxCount" : 1
+            },
+            {
+              "path" : "nsg:subRegion",
```
Comment (MFSY) [inline modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: I would suggest to merge the "nsg:subRegion" property with the "nsg:brainRegion" one.

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                "@list" : [ "3D contour", "2D contour", "1 point", "3 point" ]
+              },
+              "maxCount" : 1
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "@id" : "this:IntegrityShape",
```
Comment (MFSY) [inline modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: I would suggest to name this shape ReconstructedCellIntegerityShape.

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+              "path" : "nsg:integrity",
+              "name" : "Integrity",
+              "description" : "Gives information on the integrity of the reconstructed cell.",
+              "node" : "nsg:IntegrityShape",
+              "maxCount" : 1
+            },
+            {
+              "path" : "nsg:somaType",
+              "name" : "Soma type",
+              "description" : "Gives information on the type of some of the reconstructed cell, values can be 3D or 2D contour, 1 point, 3 point.",
```
Comment (MFSY) [inline modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: Is this list of somaType values complete ? If no, I would suggest to have a vocabulary for the values added in [typedlabeledontologyterm](https://github.com/BlueBrain/nexus-bbp-domains/blob/master/modules/bbp-core/src/main/resources/schemas/bbp/core/typedlabeledontologyterm/v0.1.0.json)

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+            },
+            {
+              "path" : "nsg:subRegion",
+              "name" : "Sub region",
+              "description" : "Sub region where the reconstructed cell is located, CA1a in hippocampus for example.",
+              "node":"bbptlotsh:BrainRegionOntologyTermShape",
+              "maxCount" : 1
+            },
+            {
+              "path" : "nsg:positionInLayer",
```
Comment (MFSY) [inline modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: The value of "nsg:positionInLayer" is not conform to [bbplocsh:BrainLocationShape](https://github.com/BlueBrain/nexus-bbp-domains/blob/master/modules/bbp-core/src/main/resources/schemas/bbp/core/brainLocation/v0.1.0.json).
I would suggest to have a property "nsg:brainLocation" defined as follows:

```json
{
              "path" : "nsg:brainLocation",
               "name" : "Brain location",
               "description" : "Location (brain region, position In Layer, longitudinalAxis) of the reconstructed cell.",
              "node":"bbplocsh:BrainLocationShape",
               "maxCount" : 1
}

Comment (MFSY) [reply modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: The solution above will allow to merge in one property all information to locate the reconstructed cell:

* positionInLayer
* longitudinalAxis
* layer

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json (hunk 6 above)
Comment (srjimenez) [reply modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: The list is complete

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json (hunk 3 above)
Comment (srjimenez) [reply modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: These are not the same thing, I think both are needed the mtype and the msubtype, they convey different information.

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json (hunk 7 above)
Comment (srjimenez) [reply modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: OK for positionInLayer and longitudinalAxis.
Layer is a field that they keep apart from the brain region and that is not as such in the parcellation schemes or Uberon. suggest to keep it apart.

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json (hunk 5 above)
Comment (srjimenez) [reply modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: OK

File: modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json (hunk 6 above)
Comment (MFSY) [reply modules/bbp-morphology/src/main/resources/schemas/bbp/morphology/reconstructedcell/v0.1.0.json]: Okay.

File: modules/bbp-morphology/src/test/resources/data/bbp/morphology/reconstruction/v0.1.0/min-fields.json [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    "{{base}}/contexts/nexus/prov/rec-prov-o-20130430/v0.1.0",
+    "{{base}}/contexts/nexus/core/standards/v0.1.0",
+    "{{base}}/contexts/bbp/neurosciencegraph/core/v0.1.0"
+  ],
+  "@type": [
+    "prov:Activity",
+    "nsg:Reconstruction"
+  ],
+  "used": [{
+    "@id": "https://bbp-nexus.epfl.ch/dev/v0/data/bbp/experiment/labeledcell/v0.1.1/8258536c-f010-467b-9c00-3c1df7a7ab11",
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/data/bbp/morphology/reconstruction/v0.1.0/min-fields.json]: Replace "https://bbp-nexus.epfl.ch/dev/v0" by {{base}}

File: modules/bbp-morphology/src/test/resources/data/bbp/morphology/reconstruction/v0.1.0/min-fields.json [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    "@type": [
+      "nsg:LabeledCell",
+      "prov:Entity"
+    ]
+  }],
+  "generated": [{
+    "@id": "https://bbp-nexus.epfl.ch/dev/v0/data/bbp/experiment/reconstructedcell/v0.1.0/90525f35-a666-47ce-962f-1d9aedf239cf",
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/data/bbp/morphology/reconstruction/v0.1.0/min-fields.json]: Replace "https://bbp-nexus.epfl.ch/dev/v0" by {{base}}

File: modules/bbp-morphology/src/test/resources/data/bbp/morphology/reconstruction/v0.1.0/min-fields.json [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    ]
+  }],
+  "startedAtTime": {
+    "@type": "xsd:dateTime",
+    "@value": "2016-08-23T00:00:00+02:00"
+  },
+  "wasStartedBy": [
+    {
+      "@type": "prov:Agent",
+      "@id": "https://bbp-nexus.epfl.ch/dev/v0/data/bbp/core/person/v0.1.0/92801f69-0db3-4945-a746-bb6abaf510e0"
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/data/bbp/morphology/reconstruction/v0.1.0/min-fields.json]: Replace "https://bbp-nexus.epfl.ch/dev/v0" by {{base}}

File: modules/bbp-morphology/src/test/resources/data/bbp/morphology/reconstruction/v0.1.0/min-fields.json [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    }
+  ],
+  "qualifiedAssociation": [{
+    "@type": "prov:Association",
+    "agent": {
+      "@type": "prov:Agent",
+      "@id": "https://bbp-nexus.epfl.ch/dev/v0/data/bbp/core/person/v0.1.0/92801f69-0db3-4945-a746-bb6abaf510e0"
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/data/bbp/morphology/reconstruction/v0.1.0/min-fields.json]: Replace "https://bbp-nexus.epfl.ch/dev/v0" by {{base}}

File: modules/bbp-morphology/src/test/resources/data/bbp/morphology/reconstruction/v0.1.0/min-fields.json [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    },
+    "hadRole": {
+      "@type": "prov:Role",
+      "@id": "HBP_ROLE:0000041"
+    }
+  }],
+  "hadProtocol": {
+    "@type": "nsg:Protocol",
+    "@id": "protocolns:e49ecf79-5d4f-454c-884f-99a17a3123e8"
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/data/bbp/morphology/reconstruction/v0.1.0/min-fields.json]: Where "protocolns" is defined.

File: modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstructedcell/v0.1.0/wrong-type.json [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+  "mType": {
+    "@id": "http://neurolex.org/wiki/Nlx_cell_20081203",
+    "label": "Thalamocortical cell"
+  },
+  "integrity": {
+    "basalDendrite": true,
+    "axon": false,
+    "soma": true
+  },
+  "somaType": "2D contour"
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstructedcell/v0.1.0/wrong-type.json]: Need a controlled vocabulary here.

File: modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstruction/v0.1.0/wrong-type.json [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+  "@context": [
+    "{{base}}/contexts/nexus/core/standards/v0.1.0",
+    "{{base}}/contexts/nexus/prov/rec-prov-o-20130430/v0.1.0",
+    "{{base}}/contexts/bbp/neurosciencegraph/core/v0.1.0"
+  ],
+  "@type": [
+    "nsg:WrongType"
+  ],
+  "used": [{
+    "@id": "https://bbp-nexus.epfl.ch/dev/v0/data/bbp/experiment/labeledcell/v0.1.1/8258536c-f010-467b-9c00-3c1df7a7ab11",
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstruction/v0.1.0/wrong-type.json]: Replace "https://bbp-nexus.epfl.ch/dev/v0" by {{base}}

File: modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstruction/v0.1.0/wrong-type.json [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+    "@type": [
+      "nsg:LabeledCell",
+      "prov:Entity"
+    ]
+  }],
+  "generated": [{
+    "@id": "https://bbp-nexus.epfl.ch/dev/v0/data/bbp/experiment/reconstructedcell/v0.1.0/90525f35-a666-47ce-962f-1d9aedf239cf",
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstruction/v0.1.0/wrong-type.json]: Replace "https://bbp-nexus.epfl.ch/dev/v0" by {{base}}

File: modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstruction/v0.1.0/wrong-type.json [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+    ]
+  }],
+  "startedAtTime": {
+    "@type": "xsd:dateTime",
+    "@value": "2016-08-23T00:00:00+02:00"
+  },
+  "wasStartedBy": [
+    {
+      "@type": "prov:Agent",
+      "@id": "https://bbp-nexus.epfl.ch/dev/v0/data/bbp/core/person/v0.1.0/92801f69-0db3-4945-a746-bb6abaf510e0"
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstruction/v0.1.0/wrong-type.json]: Replace "https://bbp-nexus.epfl.ch/dev/v0" by {{base}}

File: modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstruction/v0.1.0/wrong-type.json [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+    }
+  ],
+  "qualifiedAssociation": [{
+    "@type": "prov:Association",
+    "agent": {
+      "@type": "prov:Agent",
+      "@id": "https://bbp-nexus.epfl.ch/dev/v0/data/bbp/core/person/v0.1.0/92801f69-0db3-4945-a746-bb6abaf510e0"
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstruction/v0.1.0/wrong-type.json]: Replace "https://bbp-nexus.epfl.ch/dev/v0" by {{base}}

File: modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstruction/v0.1.0/wrong-type.json [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+    },
+    "hadRole": {
+      "@type": "prov:Role",
+      "@id": "HBP_ROLE:0000041"
+    }
+  }],
+  "hadProtocol": {
+    "@type": "Protocol",
+    "@id": "protocolns:e49ecf79-5d4f-454c-884f-99a17a3123e8"
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstruction/v0.1.0/wrong-type.json]: Define protocolns in the context.

File: modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstruction/v0.1.0/wrong-type.json [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+  "objectiveType": {
+    "@id": "http://www.bioassayontology.org/bao#BAO_0150019",
+    "label": "oil immersion lens"
+  },
+  "orientationOfReconstruction": "XY-rotated",
+  "reconstructionCompressionCorrected": "x1.25 in Z",
+  "reconstructionCompleted": true,
+  "reconstructionCorrected": false,
+  "clusterReconstructionRequested": false,
+  "reconstructionReceivedDate": {
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/reconstruction/v0.1.0/wrong-type.json]: Replace with "reconstructionReceivedDate" :"2016-08-30T00:00:00+02:00"

File: modules/bbp-morphology/src/test/resources/data/bbp/morphology/labeledcell/v0.1.0/min-fields.json [hunk 20]
```diff
@@ -11,13 +11,13 @@
...
     "label": "Primary somatosensory cortex"
   },
   "projectName": "Maria Toledo",
   "spatialCellName": "idA3",
   "putativeMType": {
-    "@id": "http://www.hbp.FIXME.org/hbp_cell_ontology/HBP_CELL:0000130",
+    "@id": "http://www.FIXME.org/cell_ontology/CELL:0000130",
     "label": "Layer IV nest basket cell"
   },
   "reconstructionRequested": true,
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/data/bbp/morphology/labeledcell/v0.1.0/min-fields.json]: To be reconsidered in a next iteration.

File: modules/bbp-core/src/main/resources/schemas/bbp/core/typedlabeledontologyterm/v0.1.0.json [hunk 21]
```diff
@@ -81,6 +81,16 @@
...
         }
       ]
+    },
+    {
+      "@id": "this:SomaTypeTermShape",
+      "@type": "sh:NodeShape",
+      "label": "The shape of an ontology term representing the types of somas of the reconstructed cell",
+      "and": [
+        {
+          "node": "bbplotsh:SomaTypeTermShape"
```
Comment (MFSY) [inline modules/bbp-core/src/main/resources/schemas/bbp/core/typedlabeledontologyterm/v0.1.0.json]: Please point to: "bbplotsh:LabeledOntologyTermShape"

File: modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/labeledcell/v0.1.0/wrong-type.json [hunk 22]
```diff
@@ -10,13 +10,13 @@
   "providerId": "C060600A3",
   "brainRegion": {
```
Comment (MFSY) [inline modules/bbp-morphology/src/test/resources/invalid/bbp/morphology/labeledcell/v0.1.0/wrong-type.json]: Shouldn't the brainRegion be part of a brainLocation object ?

**Pull request merged**